lsp/server/src/validateDocument.ts

```diff
@@ -25,25 +26,26 @@ export async function validateDocument(
     document: TextDocument,
     extensionName: string
 ): Promise<Diagnostic[]> {
-    const { uri } = document;
-
-    const setting = await getDocumentSettings(uri);
+    const setting = await getDocumentSettings(document.uri);
     let results: Diagnostic[] = [];
 
     if (document.languageId === 'javascript') {
         // handles JS rules
         const jsDiagnostics = await validateJs(document);
-        
+
```

Comment:
The result number calculation and truncating need to happen immediately after `validateJS` is called.

Reply:
Why not cull after all validation?

Reply:
If JS validator exceeds maximum, we don't need to do the GQL validator

Reply:
Maximum threshold to me is a way to cull the results array before showing them to the user. And not for early exit. If what you're saying is what we want to achieve, then, it is implied that we cull every time we call `validateXYZ(...)`. If that's the preferred approach I can change.